ATACseq_footprinting/sort.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+#SBATCH --job-name=sort
+#SBATCH --nodes=1
+#SBATCH --ntasks=1
+#SBATCH --cpus-per-task=28
+#SBATCH --partition=batch
+#SBATCH --mem=500G
+#SBATCH --time=3:00:00
+#SBATCH --account=sjaiswal
+#cd /oak/stanford/groups/sjaiswal/kameronr/JG97/methylseq/miseq/213566405/FASTQ/
+
+bam_path=$1
+
+#cd $temp_path
+
+module load samtools/1.9
+
+line_number=$SLURM_ARRAY_TASK_ID #get index of which file to process from $SLURM_ARRAY_TASK_ID provided by SLURM
+bam_file="${bam_path}/BAMs" #provide path to file containing list of fastq files
+bam_prefix="$(sed "${line_number}q; d" "${bam_file}")" #extract only the line number corresponding to $SLURM_ARRAY_TASK_ID
+
+FILENAME=$(basename "${bam_prefix}")
+PREFIX=$FILENAME
+echo "filename: $PREFIX"
+
+cd $bam_path
+number_replicates=$(find . -name "${PREFIX}_Rep[1-9]*_treat_rep1.bam" |
+            sed "s/${PREFIX}_Rep\([0-9][0-9]*\)\_treat_rep1.bam/\1/" |
+            sort -n |
+            tail -n 1)
+reps=$(basename "${number_replicates}")
+
+echo "number of replicates for ${PREFIX}: $reps"
+
+if [ ! -f "$bam_path/${PREFIX}_Rep1_treat_rep1.sorted.bam" ]; then
```

Comment:
I think we don't want to sort the bams that have replicates until after they are merged. For the bams that don't have replicates, they don't need merging, so they can just be sorted at the beginning. 

If merging is faster on sorted bams, then you could sort everything as first step.

Reply:
We have to sort the bams before merging them. The samtools merge command won't work on unsorted bams. After they are merged, they have to be sorted again in order to be indexed (as the samtools merge command does not perform any sorting)